### PR TITLE
Feature/jc19 487 displaying details for a selected anime [React]

### PR DESCRIPTION
**Pull request by andrey-gazzaev**
#JC19-487

### CODE REVIEW

File: apps/react/src/features/anime/components/Details/Details.module.css [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+.details .loader {
+  position: absolute;
+  top: 50%;
+  right: 50%;
+  transform: translate(50%, -50%);
+}
+
+.details .placeholderTest {
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/features/anime/components/Details/Details.module.css:43]: I think you made a typo.

File: apps/react/src/features/anime/components/ImagePopup/ImagePopup.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const ImagePopupComponent: FC<Props> = ({ isOpen, onClose, imageSrc }) => {
+
+  /** Handles close dialog window. */
+  const handleClose = useCallback(() => {
+    onClose();
+  }, []);
+
+  return (
+    <Dialog onClose={handleClose} open={isOpen}>
+      <img src={imageSrc} alt="" />
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/features/anime/components/ImagePopup/ImagePopup.tsx:25]: I think you should provide `alt` with a prop

**Review by artem-sotnichenko-saritasa — APPROVED** [review]
https://saritasa-nest.github.io/camp-js-2022-andrey-gazzaev/pr-preview/pr-17/#/anime?search=&types=TV&field=english&direction=des&id=10929
On my screen, youtube player is cropped and I can't scroll.

File: apps/react/src/features/anime/components/Details/Details.tsx [hunk 3]
```diff
@@ -1,15 +1,198 @@
...
+          className={styles.placeholderText}
+        >
+          Please select an anime
+        </Typography>
+      </Container>
+    );
+  }
+
+  const airedStart = animeDetails.aired.start === null ?
+    'no date' :
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime/components/Details/Details.tsx:97]: no date repeats many times, let's move it con const.

File: apps/react/src/api/services/animeService.ts [hunk 4]
```diff
@@ -49,11 +51,27 @@ export namespace AnimeService {
     return [...results];
   }
 
+  /**
+   * Fetches anime by id.
+   * @param id Anime Id.
+   */
+  export async function fetchAnimeById(id: number): Promise<AnimeDetails | null> {
+    try {
+      const url = `${animeUrl.toString()}${id}/`;
```
Comment (AndreySkripachev) [inline apps/react/src/api/services/animeService.ts:60]:
```
new URL(id, animeUrl.toString())
```

File: apps/react/src/features/anime/components/AnimeList/AnimeList.tsx [hunk 5]
```diff
@@ -54,29 +56,39 @@ const getAnimeListOptions = (searchParams: URLSearchParams): AnimeListQueryParam
...
   };
 };
 
 const AnimeListComponent: FC<Props> = () => {
   const animeList = useAppSelector(selectAmineList);
   const isLoading = useAppSelector(selectIsAnimeLoading);
   const [searchParams, setSearchParams] = useSearchParams();
-  const [query, setQuery] = useState<AnimeListQueryParams>(getAnimeListOptions(searchParams));
+  const [query, setQuery] = useState<AnimeListQueryParamsWithId>(getAnimeListOptions(searchParams));
+  const [currentAnimeId, setCurrentAnimeId] = useState(query.id);
```
Comment (AndreySkripachev) [inline apps/react/src/features/anime/components/AnimeList/AnimeList.tsx:68]: Is it really necessary?

File: package.json [hunk 6]
```diff
@@ -37,16 +37,19 @@
     "@ngx-env/core": "^2.1.1",
     "@reduxjs/toolkit": "^1.8.2",
     "axios": "^0.27.2",
+    "clsx": "^1.2.1",
     "core-js": "^3.6.5",
     "env-cmd": "^10.1.0",
     "formik": "^2.2.9",
     "formik-mui": "^5.0.0-alpha.0",
+    "moment": "^2.29.4",
```
Comment (chernodub) [inline package.json:45]: Moment is considered to be deprecated, please prefer some more modern solutions
https://momentjs.com/docs/#/-project-status/

**Review by chernodub — CHANGES_REQUESTED** [review]

**Pull request closed without merging**